### PR TITLE
feat(pwa): make prefetch job args configurable via flux

### PR DESCRIPTION
**Pull request by philippsauer**
## PR Type

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] CI-related changes
- [ ] Documentation content changes
- [ ] Application / infrastructure changes

## Release ##

Be sure that pull requests are build according to the defined release process [here](https://github.com/intershop/helm-charts/wiki/Release-Process). As a main part to mention here is that the semantic version type will be read from the commit messages (`BREAKING CHANGE(icm):` marks a *major* change, `feat(icm):` marks *minor* changes and the rest will be *patch*. So the developer must already know and is responsible.

## What Is the Current Behavior?

The arguments for the prefetch job were hard coded in the helm chart

## What Is the New Behavior?

The arguments for the prefetch job can now be configured via values.yaml

## Does this PR Introduce a Breaking Change?

No breaking change, if no args are provided, the default settings will apply (same as before the change)

- [ ] Yes
- [n] No

### CODE REVIEW

File: charts/pwa/templates/prefetch-job.yaml [hunk 1]
```diff
@@ -27,15 +27,7 @@ spec:
...
-              "--timeout=15",
-              "--spider",
-              "--no-check-certificate",
-              "--retry-connrefused",
-              "--tries=5",
-              "--execute=robots=off",
-              "--recursive",
-              "--level=0",
-              "--no-verbose",
+              {{ default "\"--timeout=15\", \"--spider\", \"--no-check-certificate\", \"--retry-connrefused\", \"--tries=5\", \"--execute=robots=off\", \"--recursive\", \"--level=0\", \"--no-verbose\"" .args }},
```
Comment (philippsauer) [inline charts/pwa/templates/prefetch-job.yaml:30]: Providing a multiline string here would remove the indention of each line in the resulting file, so I decided to keep it in a single line. Maybe there could be a more beautiful solution

File: charts/pwa/values.schema.json [hunk 2]
```diff
@@ -170,6 +170,11 @@
                       "type": "integer",
                       "title": "The maximum duration (in seconds) of a running prefetch job",
                       "default": "3600"
+                    },
+                    "args": {
+                      "$id": "#/properties/cache/properties/prefetch/properties/args",
+                      "type": "string",
+                      "title": "The arguments for the wget job"
```
Comment (philippsauer) [inline charts/pwa/values.schema.json:177]: Alternatively the default could be added here, but same problem as above.

File: charts/pwa/templates/prefetch-job.yaml (hunk 1 above)
Comment (shauke) [reply charts/pwa/templates/prefetch-job.yaml:30]: Unfortunately I currently have no better idea. So I will merge it for testing as it is.

File: charts/pwa/values.schema.json (hunk 2 above)
Comment (shauke) [reply charts/pwa/values.schema.json:177]: No better idea either. ;)

**Review by shauke — APPROVED** [review]

**Pull request merged**